src/main/java/com/google/sps/model/AuthenticatedHttpServlet.java

```diff
@@ -129,29 +136,34 @@ public void doPost(
    * HTTP Request
    *
    * @param request Http Request sent from client
-   * @return Credential object with accessToken. null if tokens not present / are empty
-   * @throws TokenVerificationError if the passed idToken is confirmed to be false / is unverifiable
+   * @return Credential object with accessToken.
+   * @throws CredentialVerificationException if the credentials are invalid / not present
    * @throws GeneralSecurityException if an issue occurs with Google's verification service
    * @throws IOException if an issue occurs with Google's verification service
    */
   private Credential getGoogleCredential(HttpServletRequest request)
-      throws TokenVerificationError, GeneralSecurityException, IOException {
-    // If cookies not present, return null (user's session likely expired)
-    Cookie idTokenCookie = ServletUtility.getCookie(request, "idToken");
-    Cookie accessTokenCookie = ServletUtility.getCookie(request, "accessToken");
-    if (idTokenCookie == null || accessTokenCookie == null) {
-      return null;
+      throws CredentialVerificationException, GeneralSecurityException, IOException {
+    Cookie idTokenCookie;
+    Cookie accessTokenCookie;
+
+    try {
+      idTokenCookie = ServletUtility.getCookie(request, "idToken");
+    } catch (CookieParseException e) {
+      throw new CredentialVerificationException("idToken is not present / cannot be parsed!");
```

Comment:
Whenever you catch an exception and throw a different exception, you should make the original exception the "cause" or "inner exception". All built-in exception classes should have a constructor that include an Exception argument after the message string. Your own exceptions should have these constructors as well.

Reply:
This is cool! Implemented this no problem - seems helpful for keeping a chain of how exceptions were caused (which I always benefit from when tests fail / I'm debugging). Thanks!